scripts/rollup/shims/react-native/ReactNativeViewConfigRegistry.js

```diff
@@ -19,6 +19,7 @@ const customBubblingEventTypes: {
     phasedRegistrationNames: $ReadOnly<{|
       captured: string,
       bubbled: string,
+      skipBubbling?: ?boolean,
     |}>,
```

Comment:
This type implies the possibility of a nonsensical value:

```
phasedRegistrationNames: {
  captured: 'onFooCapture',
  bubbled: 'onFooWhateverIFeelLike', // Not actually used, so why does it matter?
  skipBubbling: true,
},
```

Could you have accomplished the same by making `bubbled` nullable?
```suggestion
    phasedRegistrationNames: $ReadOnly<{|
      captured: string,
      bubbled: null | string,
    |}>,
```

Reply:
Yea I thought about this but we still need a way to define what the bubbled event listener name is. Because even if we don't bubble, we still need to dispatch that listener on the `target`. Unless we can infer the name of event by the `captured` listener?

Reply:
I guess clarification -- the bubbled listener is still dispatched on the target even if the event is non-bubbling

Reply:
Ahh, right. This really reveals how terrible the current event config is at describing the events, heh. Thanks for the explanation.

Reply:
I may be missing some previous discussions on this, but what do we think about calling this `bubbles` to bring it in line with the name [used on the Web](https://javascript.info/dispatch-events#bubbling-example) for custom events?

Long term, the event config could match even closer to the Web spec for dispatching custom events, something like:

```js
{
  type: string,
  bubbles: boolean,
  
  // Not in web spec, but could be implemented in React.
  captures: boolean,
}
```

Which would turn this:

```js
{
  bubblingEventTypes: {
    topDefaultBubblingEvent: {
      phasedRegistrationNames: {
        captured: 'onDefaultBubblingEventCapture',
        bubbled: 'onDefaultBubblingEvent',
      },
    },
  }
  directEventTypes: {
    topDirectEvent: {
       registrationName: 'onDirectEvent',
    },
  }
}
```

Into this:

```js
[
  {
    // Bubbling phase is onDefaultBubblingEvent
    // Capture phase is onDefaultBubblingEventCapture
    // Top is topDefaultBubblingEvent
    // Captures and bubbles
    type: 'DefaultBubblingEvent',
    bubbles: true,
    captures: true,
  },
  {
    // Direct event name is onDirectEvent, no capture, no bubble.
    type: 'DirectEvent',
    bubbles: false,
    captures: false,
  }
]
```

Notice that there's only one config (instead of bubbling vs direct), and all of the event names like `top*`, `on*`, and `on*Capture` are automatically derived from the event type, similar to what we do in React DOM with things like `onClick` and `onClickCapture`.